ed2nav: fix bug with lacking edges on street network

kinnou02 opened the pull request:
When choosing nodes to be ignored in the graph we were using the
vertex_t of the node and not it's id, so we were deleting the wrong
node, that caused some edge to be deleted too.

Ironically when removing small connected component we were in fact
randomly created some of them.

Refer to: http://jira.canaltp.fr/browse/NAVITIAII-1691

kinnou02 committed:
ed2nav: fix bug with lacking edges on street network

When choosing nodes to be ignored in the graph we were using the
vertex_t of the node and not it's id, so we were deleting the wrong
node, that caused some edge to be deleted too.

Ironically when removing small connected component we were in fact
randomly created some of them.

antoine-de commented:
rah you got the thousandth ! :sob:

TeXitoi commented on source/georef/georef.h:
```diff
@@ -151,6 +151,16 @@ struct Way :public nt::Nameable, nt::Header{
+    template <typename Stream, typename G>
+    void print(Stream& stream, const G& g){
+        for(auto& edge: this->edges){
+            stream << "LINESTRING(" << g[edge.first].coord.lon() << " " << g[edge.first].coord.lat()
```
you can add `stream << std::setprecision(16) << "LINESTRING...` to have a correct precision.

TeXitoi commented on source/ed/ed_reader.cpp:
```diff
@@ -1078,8 +1079,9 @@ void EdReader::fill_vector_to_ignore(navitia::type::Data& , pqxx::work& work,
         BOOST_FOREACH(navitia::georef::edge_t e, boost::out_edges(vertex_idx, geo_ref_temp.graph)) {
             uint64_t target = boost::target(e, geo_ref_temp.graph);
             edge_to_ignore.insert({source, target});
```
source is a "osm_idex" and target is a "node_descriptor", I can't understand how it can work. And osm_idex is a very strange name.

kinnou02 replied:
it won't work :) or more precisely, it work because it don't do anything...

kinnou02 committed:
Ed2nav: better handling of edge_to_ignore

Since we don't import node not required by the graph, we don't have to
known the edge we don't have to import, it won't be possible to create
them anyway.

kinnou02 committed:
kraken: set precision before exporting graph in debug

antoine-de commented on source/ed/ed_reader.cpp:
```diff
@@ -1072,13 +1072,12 @@ void EdReader::fill_vector_to_ignore(navitia::type::Data& , pqxx::work& work,
             continue; //big enough, we skip
 
-        uint64_t source = osmid_idex[vertex_idx];
+        uint64_t source = map_idx_to_id[vertex_idx];
```
why don't you use the edge_to_ignore anymore ? 
If it is not useful anymore, remove it
